air-script/tests/functions/functions_complex.rs

```diff
@@ -86,6 +86,6 @@ impl Air for FunctionsAir {
         let main_next = main_frame.next();
         let aux_current = aux_frame.current();
         let aux_next = aux_frame.next();
-        result[0] = aux_next[0] - aux_current[0] * ((E::from(main_current[3]).exp(E::PositiveInteger::from(4_u64)) * E::from(main_current[1]) * E::from(main_current[2]) + E::from(main_current[3]).exp(E::PositiveInteger::from(2_u64)) * (E::ONE - E::from(main_current[1])) * E::from(main_current[2]) + E::from(main_current[3]) * E::from(main_current[1]) * (E::ONE - E::from(main_current[2])) + (E::ONE - E::from(main_current[1])) * (E::ONE - E::from(main_current[2]))) * E::from(main_current[0]) - E::from(main_current[0]) + E::ONE);
+        result[0] = aux_next[0] - aux_current[0] * ((E::from(main_current[3]) * E::from(main_current[3]) * E::from(main_current[3]) * E::from(main_current[3]) * E::from(main_current[3]) * E::from(main_current[3]) * E::from(main_current[3]) * E::from(main_current[1]) * E::from(main_current[2]) + E::from(main_current[3]) * E::from(main_current[3]) * (E::ONE - E::from(main_current[1])) * E::from(main_current[2]) + E::from(main_current[3]) * E::from(main_current[1]) * (E::ONE - E::from(main_current[2])) + (E::ONE - E::from(main_current[1])) * (E::ONE - E::from(main_current[2]))) * E::from(main_current[0]) - E::from(main_current[0]) + E::ONE);
```

Comment:
Not for this PR, but I think the ideal output here would be something like:
```rust
let tmp1 = main_current[3] * main_current[3]; // main_current[3]^2
let tmp2 = tmp1 * tmp1;                       // main_current[3]^4
let tmp3 = F::ONE - main_current[1];
let tmp4 = F::ONE - main_current[2];

result[0] = aux_next[0] - aux_current[0] * E::from(
    (
        main_current[3] * tmp2 * tmp1 * main_current[1] * main_current[2]
        + tmp2 * tmp3 * main_current[2]
        + main_current[3] * main_current[1] * tmp4 
        + tmp3 * tmp4
    )
    * main_current[0] - main_current[0] + F::ONE
);
```
The main points being:
- We introduce temporary variables to hold results of common sub-expressions.
- We delay conversion to the extension filed to the end.

Reply:
Issue created here: https://github.com/0xPolygonMiden/air-script/issues/354